src/server/pps/server/api_server.go

```diff
@@ -95,12 +74,62 @@ func (a *apiServer) CreateJob(ctx context.Context, request *ppsclient.CreateJobR
 	if len(repoSet) < len(request.Inputs) {
 		return nil, fmt.Errorf("pachyderm.ppsclient.jobserver: duplicate repo in job")
 	}
+
+	jobID := uuid.NewWithoutDashes()
+
+	var parentJobInfo *persist.JobInfo
+	var err error
+	if request.ParentJob != nil {
+		inspectJobRequest := &ppsclient.InspectJobRequest{Job: request.ParentJob}
+		parentJobInfo, err = a.persistAPIServer.InspectJob(ctx, inspectJobRequest)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	reduce := false
+	if parentJobInfo != nil {
+		for _, jobInput := range parentJobInfo.Inputs {
+			if jobInput.Reduce {
+				reduce = true
+			}
+		}
+	}
```

Comment:
I'm not sure either.  This code seems to have the effect that, if a job has a parent, and its parent is a reduce job, then create a separate repo for this job (instead of using its parent's repo).  Do we want that?